app/src/main/java/com/scouts/kitchenplaner/ui/view/createproject/AllergenAdder.kt

```diff
@@ -41,50 +45,66 @@ import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
 import com.scouts.kitchenplaner.ui.state.AllergenPersonAdderState
 import com.scouts.kitchenplaner.ui.view.DockedDatePicker
 
 @Composable
-fun AllergenAdder(state: AllergenPersonAdderState, onAdd: () -> Unit) {
-    val columnItemModifier = Modifier
-        .padding(5.dp)
-        .height(70.dp)
-    Column (
-        modifier = Modifier.verticalScroll(rememberScrollState())
-    ) {
-        AllergenInputs(state, columnItemModifier)
+fun AllergenAdder(state: AllergenPersonAdderState, onAdd: () -> Unit, onDismiss: () -> Unit) {
+    Dialog(onDismissRequest = onDismiss) {
+        Surface(
+            modifier = Modifier.fillMaxHeight(0.5f),
+            shape = RoundedCornerShape(16.dp),
+            color = MaterialTheme.colorScheme.background
+        ) {
+            val columnItemModifier = Modifier
+                .padding(5.dp)
+                .height(70.dp)
+            Column(
+                modifier = Modifier
+                    .verticalScroll(rememberScrollState())
+            ) {
+                AllergenInputs(state, columnItemModifier)
 
-        HorizontalDivider()
+                HorizontalDivider()
 
-        AllergenAddInputs(
-            onAdd = { allergen, traces -> state.addAllergen(allergen, traces) },
-            modifier = Modifier.padding(10.dp)
-        )
-
-        if (state.allergens.isNotEmpty()) {
-            HorizontalDivider()
+                AllergenAddInputs(
```

Comment:
- You can add "" as allergen and you can add the same string multiple times
- You can add an allergen with and without trace (s.t. it appears twice)